src/handlers/gestures/useGesture.ts

```diff
@@ -0,0 +1,215 @@
+        gst.handlerTag,
+        filterConfig(gst.config, ALLOWED_PROPS)
+      );
+
+      registerHandler(gst.handlerTag, gst);
+
+      setImmediate(() => {
+        gestureConfig.prepare();
+
+        let requireToFail: number[] = [];
+        if (gst.config.requireToFail) {
+          requireToFail = gst.config.requireToFail
+            .map(convertToHandlerTag)
+            .filter((tag) => tag > 0);
+        }
+
+        let simultaneousWith: number[] = [];
+        if (gst.config.simultaneousWith) {
+          simultaneousWith = gst.config.simultaneousWith
+            .map(convertToHandlerTag)
+            .filter((tag) => tag > 0);
+        }
+
+        RNGestureHandlerModule.updateGestureHandler(
+          gst.handlerTag,
+          filterConfig(gst.config, ALLOWED_PROPS, {
+            simultaneousHandlers: simultaneousWith,
+            waitFor: requireToFail,
+          })
+        );
+      });
+    }
+    result.current.config = gesture;
+    result.current.callback?.();
+
+    if (result.current.animatedHandlers) {
+      result.current.animatedHandlers.value = (gesture.map(
+        (g) => g.handlers
+      ) as unknown) as HandlerCallbacks<Record<string, unknown>>[];
+    }
+  }
+
+  function updateHandlers() {
+    gestureConfig.prepare();
+
+    for (let i = 0; i < gesture.length; i++) {
+      const gst = result.current.config[i];
+
+      gesture[i].handlerTag = gst.handlerTag;
+      gesture[i].handlers.handlerTag = gst.handlerTag;
+    }
+
+    for (let i = 0; i < gesture.length; i++) {
+      const gst = result.current.config[i];
+
+      gst.config = gesture[i].config;
+      gst.handlers = gesture[i].handlers;
+      gst.handlers.handlerTag = gst.handlerTag;
+
+      let requireToFail: number[] = [];
+      if (gst.config.requireToFail) {
+        requireToFail = gst.config.requireToFail
+          .map(convertToHandlerTag)
+          .filter((tag) => tag > 0);
+      }
+
+      let simultaneousWith: number[] = [];
+      if (gst.config.simultaneousWith) {
+        simultaneousWith = gst.config.simultaneousWith
+          .map(convertToHandlerTag)
+          .filter((tag) => tag > 0);
+      }
+
+      RNGestureHandlerModule.updateGestureHandler(
+        gst.handlerTag,
+        filterConfig(gst.config, ALLOWED_PROPS, {
+          simultaneousHandlers: simultaneousWith,
+          waitFor: requireToFail,
+        })
+      );
+
+      registerHandler(gst.handlerTag, gst);
+    }
+
+    if (result.current.animatedHandlers) {
+      result.current.animatedHandlers.value = (result.current.config.map(
+        (g) => g.handlers
+      ) as unknown) as HandlerCallbacks<Record<string, unknown>>[];
+    }
+  }
+
+  function needsToReattach() {
+    if (gesture.length !== result.current.config.length) {
+      return true;
+    } else {
+      for (let i = 0; i < gesture.length; i++) {
+        if (gesture[i].handlerName !== result.current.config[i].handlerName) {
+          return true;
+        }
+      }
+    }
```

Comment:
You don't need `else` here.

Reply:
Removed in 0acba7d437cd505513529f02eeca5d5610bba6b5